fpdf/fpdf.py

```diff
@@ -2156,29 +2199,39 @@ def _render_styled_cell_text(
                     f"{(x + w) * k:.2f} {(self.h - (y + h)) * k:.2f} l S "
                 )
 
-        if styled_txt_frags:
+        s_start = self.x
+        s_width, underlines = 0, []
+        if text_line.fragments:
             if align == "R":
                 dx = w - self.c_margin - styled_txt_width
             elif align == "C":
                 dx = (w - styled_txt_width) / 2
             else:
                 dx = self.c_margin
+            s_start += dx
 
             if self.fill_color != self.text_color:
                 s += f"q {self.text_color} "
 
             prev_font_style, prev_underline = self.font_style, self.underline
-            s_width, underlines = 0, []
             s += (
                 f"BT {(self.x + dx) * k:.2f} "
                 f"{(self.h - self.y - 0.5 * h - 0.3 * self.font_size) * k:.2f} Td"
             )
-            # If multibyte, Tw has no effect - do word spacing using an
-            # adjustment before each space
-            if self.ws and self.unifontsubset:
+
+            word_spacing = 0
```

Comment:
What is the relation of this variable with `self.ws`?
Could a short comment be added here please, explaning things a little?

Reply:
will do.